provider/fs/src/export/serializers/mod.rs

```diff
@@ -6,42 +6,18 @@ pub mod bincode;
 pub mod json;
 pub mod postcard;
 
-use displaydoc::Display;
 use icu_provider::buf::BufferFormat;
+use icu_provider::DataError;
 use std::io;
 
-/// An Error type specifically for the [`Serializer`](serde::Serializer) that doesn't carry filenames
-#[derive(Display, Debug)]
-#[non_exhaustive]
-pub enum Error {
```

Comment:
Thought: IIRC, one motivation of having separate internal error types like this one was to prevent accidentally forgetting to associate the path with the error, since in order to convert from `serializers::Error` to `DataError` you needed to attach a path. But it's a lot of boilerplate to achieve that vague objective, so I am supportive of this simplication.